nilearn/decomposition/base.py

```diff
@@ -4,22 +4,60 @@
 """
 from __future__ import division
 from math import ceil
-
 import itertools
+from distutils.version import LooseVersion
+
 import numpy as np
+
 from scipy import linalg
+import sklearn
 from sklearn.base import BaseEstimator
 from sklearn.externals.joblib import Memory, Parallel, delayed
 from sklearn.linear_model import LinearRegression
 from sklearn.utils import check_random_state
-from sklearn.utils.extmath import randomized_svd
+from sklearn.utils.extmath import randomized_svd, svd_flip
 from .._utils.cache_mixin import CacheMixin, cache
 from .._utils.niimg import _safe_get_data
 from .._utils.compat import _basestring
 from ..input_data import NiftiMapsMasker
 from ..input_data.masker_validation import check_embedded_nifti_masker
 
 
+def fast_svd(X, n_components, random_state=None):
+    """ Automatically switch between randomized and lapack SVD (heuristic
+        of scikit-learn).
+    """
```

Comment:
I think that documentation could be useful and not supposed to a private function. Could you please add documentation ?